scripts/diagnostics/afn_auditor.py

```diff
@@ -0,0 +1,400 @@
+                afn_zone = None
+                for _, node in self.nodes.items():
+                    if node['zone_name'] == zone_name:
+                        afn_zone = node
+                        node['link_count'] += 1
+                        node['external_connections'] += 1
+                        if surf['external_node_name'] in node['neighbors']:
+                            node['neighbors'][surf['external_node_name']] += 1
+                        else:
+                            node['neighbors'][surf['external_node_name']] = 1
+                        if zone_name in external_node['neighbors']:
+                            external_node['neighbors'][zone_name] += 1
+                        else:
+                            external_node['neighbors'][zone_name] = 1
+                        break
+                if afn_zone is None:
+                    raise auditor.BadModel('Failed to find AirflowNetwork zone for thermal zone "' + zone_name + '"')
+                linked_nodes = [afn_zone, external_node]
+            elif bc == 'Surface':
+                zone_name = ht_surf['zone_name']
+                # Find the multizone zone that points at this zone
+                afn_zone = None
+                for _, node in self.nodes.items():
+                    if node['zone_name'] == zone_name:
+                        afn_zone = node
+                        node['link_count'] += 1
+                        break
+                if afn_zone is None:
+                    raise auditor.BadModel('Failed to find AirflowNetwork zone for thermal zone "' + zone_name + '"')
+                linked_nodes = [afn_zone]
+                adjhtsurf = ht_surfs[ht_surf['outside_boundary_condition_object']]
+                adj_zone_name = adjhtsurf['zone_name']
+                adj_afnzone = None
+                for _, node in self.nodes.items():
+                    if node['zone_name'] == adj_zone_name:
+                        adj_afnzone = node
+                        node['link_count'] += 1
+                        break
+                if adj_afnzone is None:
+                    raise auditor.BadModel(
+                        'Failed to find AirflowNetwork zone for adjacent thermal zone "' + adj_zone_name + '"'
+                    )
+                linked_nodes.append(adj_afnzone)
+                if adj_zone_name in afn_zone['neighbors']:
+                    afn_zone['neighbors'][adj_zone_name] += 1
+                else:
+                    afn_zone['neighbors'][adj_zone_name] = 1
+                if zone_name in adj_afnzone['neighbors']:
+                    adj_afnzone['neighbors'][zone_name] += 1
+                else:
+                    adj_afnzone['neighbors'][zone_name] = 1
+                
+            surf['nodes'] = linked_nodes
+        return True
+
+    def audit(self, **kwargs):
+        if self.nodes == {} or self.external_nodes == {} or self.surfs == {}:
+            self.extract()
```

Comment:
The IDE is complaining that it can't find these extract and connect methods.  I'm assuming they need to be the underscore prefixed versions, but I just want to confirm.

Reply:
@Myoldmopar All these changes are making the merge situation I'm looking at quite a lot more difficult. I'll be reverting some of the variable name changes.